docs/tutorials/tutorial-1.md

```diff
@@ -0,0 +1,766 @@
+
+Fortunately, TSA has a mechanism to model this precise situation. Just surround all your preconditions between the functions `tsa_forbid_failures` and `tsa_allow_failures`:
+
+```c
+() main() impure {
+    
+    ;; Signal TSA to ignore failures happening after this function. 
+    ;; If a failure happens, TSA will try another instantiation,
+    ;; so that only instantiations that successfully execute all
+    ;; pre-conditions will reach the tsa_allow_failures function call below.
+    tsa_forbid_failures();
+
+    ;; Each pre-condition could fail
+
+    pre_condition_1
+
+    pre_condition_2
+
+    pre_condition_3
+
+    .... etc
+
+    ;; Signal TSA to stop if a failure happens after this function.
+    tsa_allow_failures();
+
+    throw_if(..., assertion_to_check);
+}
+```
+
+Function `tsa_forbid_failures` tells TSA to ignore any failures happening after the function call. If a failure happens after the call to `tsa_forbid_failures`, TSA will try a different instantiation of the symbolic variables until it finds an instantiation that successfully reaches the call to `tsa_allow_failures`, which activates again the normal behavior of TSA: stop at any failure. So that when TSA reaches `throw_if(..., assertion_to_check)`, if `assertion_to_check` turns out to be true, `throw_if` will throw the exception and TSA will issue a report with the instantiations to the symbolic variables found so far.
+
+With the above explanation, we are ready to write the checker contract. The assertion we want to check is:
+
+> There is a message `m` such that: we can successfully send `m` to the PeekLevel contract, AND immediately afterwards we can successfully invoke the `locked` getter to obtain the value of the `locked` field, AND the `locked` field's value is `false`.
+
+In `/home/tutorial-1`, create a file named `checker.fc`. Paste the following code and save it. This checker contract encodes the above assertion. The explanation of how this is accomplished is given after the code:
+
+```c
+#include "imports/stdlib.fc";
+#include "imports/tsa_functions.fc";
+
+() locked() method_id { }
+
+int id_locked() asm "locked PUSHINT";
+
+() main() impure {
+
+    tsa_forbid_failures();
+        
+    ;; Send a message to the PeekLevel contract (identified with index 1).
+    ;; Show the message in the final report under name "0".
+    tsa_send_internal_message(1, 0);
+
+    ;; Call the locked getter in the PeekLevel contract.
+    ;; The result is the value of the locked field.
+    ;; Store the result in locked_val.
+    int locked_val = tsa_call_1_0(1, id_locked());
+            
+    tsa_allow_failures();
+    
+    ;; Stop if locked_val is false
+    throw_if(256, locked_val == 0);
+}
+```
```

Comment:
Maybe also mention in this section why we don't need to set any preconditions here?